packages/kbn-openapi-generator/src/template_service/templates/zod_operation_schema.handlebars

```diff
@@ -31,8 +31,17 @@ export type {{@key}} = {{> ts_type}};
 export type {{@key}}Input = {{> ts_input_type }};
 export const {{@key}}: z.ZodType<{{@key}}, ZodTypeDef, {{@key}}Input> = {{> zod_schema_item }};
 {{else}}
+{{#if (shouldCastExplicitly this)}}
+// We need this temporary type to infer from it below, but in the end we want to export as a casted {{@key}} type
```

Comment:
Comment reveals implementation details in each generated file. It doesn't look necessary here and may only confuse readers. I recommend to transform it to a handlebars comment or remove completely.